include/qengine_opencl.hpp

```diff
@@ -35,33 +32,49 @@ typedef std::shared_ptr<QEngineOCL> QEngineOCLPtr;
 /** OpenCL enhanced QEngineCPU implementation. */
 class QEngineOCL : public QEngineCPU {
 protected:
+    int deviceID;
     OCLEngine* clObj;
+    DeviceContextPtr device_context;
     cl::CommandQueue queue;
-    cl::Buffer stateBuffer;
+    cl::Context context;
+    BufferPtr stateBuffer;
```

Comment:
Why is this one a BufferPtr but the rest of them aren’t?

Reply:
This is the only buffer that's potentially modified outside of the instance itself, specifically by QEngineOCLMulti. The others never will be.